script.js

```diff
@@ -23,17 +23,32 @@ const addTodoHandler = () => {
     renderTodos(todos);
 };
 
+const deleteTodoHandler = (todoId) => {
+    todos = [...todos].filter((todo) => todo.id !== todoId);
+
+    renderTodos(todos);
+};
+
 const createTodoElement = (todo) => {
     const $todo = document.createElement("li");
+    const $deleteButton = document.createElement("button");
+    $deleteButton.innerText = "Delete";
+
+    eventListenerHandler($deleteButton, "click", () =>
+        deleteTodoHandler(todo.id)
+    );
+
     $todo.innerHTML = `<p>${todo.title}</p>`;
+    $todo.appendChild($deleteButton);
 
-    return $todo.outerHTML;
+    return $todo;
 };
 
 const renderTodos = (todos) => {
     $todoList.innerHTML = "";
-
-    $todoList.innerHTML = todos.map((todo) => createTodoElement(todo)).join("");
+    todos.forEach((todo) => {
+        $todoList.appendChild(createTodoElement(todo));
+    });
```

Comment:
Why are we opting back to previous suggestion?

Reply:
As our createElement create new element. If we use the map method to render the todos it just return us Object html Li element. so I had to make it  to string using outerHTml. But  when I dynamically added button and eventListeners, it also  converted into string. So finally I can't find the events. For this we have to use forEach method. As each time we will append a new child by this. Also we don't have to use outerHTML too.